purchase_stock_picking_return_invoicing/models/purchase_order.py

```diff
@@ -127,15 +127,29 @@ def _compute_qty_refunded(self):
         'move_ids.returned_move_ids.state',
     )
     def _compute_qty_returned(self):
-        for line in self:
-            qty = 0.0
-            moves = line.mapped('move_ids.returned_move_ids')
-            for move in moves.filtered(lambda x: x.state == 'done'):
-                if move.location_id.usage != 'supplier':
-                    qty += move.product_uom._compute_quantity(
-                        move.product_uom_qty, line.product_uom,
-                    )
-            line.qty_returned = qty
+        """Made through read_group for not impacting in performance."""
+        StockMove = self.env['stock.move']
+        origin_moves = StockMove.search([
+            ('origin_returned_move_id.purchase_line_id', 'in', self.ids),
+        ])
+        groups = StockMove.read_group(
+            [('id', 'in', origin_moves.ids),
```

Comment:
I think you can put here directly `('origin_returned_move_id.purchase_line_id', 'in', self.ids)` and remove the previous search.